src/fluree/db/json_ld/credential.cljc

```diff
@@ -77,35 +77,38 @@
-                                      (jld-processor/canonize)
-                                      (crypto/sha2-256))
-                             :cljs (<p! (-> (get credential "credentialSubject")
-                                            (jld-processor/canonize)
-                                            (.then (fn [res] (crypto/sha2-256 res))))))
-
-            proof-did     (get-in credential ["proof" "verificationMethod"])
-            pubkey        (did/decode-did-key proof-did)]
-        (when (not= jws-header-json header)
-          (throw (ex-info "Unsupported jws header in credential."
-                          {:error :credential/unknown-signing-algorithm
-                           :supported-header jws-header-json
-                           :header header
-                           :credential credential})))
-
-        (when (not (crypto/verify-signature pubkey signing-input signature))
-          (throw (ex-info "Verification failed." {:error :credential/invalid-signature :credential credential})))
-        ;; everything is good
-        true)
-      (catch* e
-              (throw (ex-info "Unverifiable credential"
-                              {:credential credential
-                               :error :credential/unverifiable
-                               :message e}))))))
+    (if-let [jws (get-in credential ["proof" "jws"])]
+      (try*
+        (let [subject (get credential "credentialSubject")
+              issuer  (get credential "issuer")
+              {:keys [header signature]} (deserialize-jws jws)
+
+              signing-input #?(:clj (-> (jld-processor/canonize subject)
+                                        (crypto/sha2-256))
+                               :cljs (<p! (-> (jld-processor/canonize subject)
+                                              (.then (fn [res] (crypto/sha2-256 res))))))
+
+              proof-did     (get-in credential ["proof" "verificationMethod"])
+              pubkey        (did/decode-did-key proof-did)]
+          (when (not= jws-header-json header)
+            (throw (ex-info "Unsupported jws header in credential."
+                            {:error :credential/unknown-signing-algorithm
+                             :supported-header jws-header-json
+                             :header header
+                             :credential credential})))
+
+          (when (not (crypto/verify-signature pubkey signing-input signature))
+            (throw (ex-info "Verification failed." {:error :credential/invalid-signature :credential credential})))
+          ;; everything is good
+          {:subject subject :issuer issuer})
+        (catch* e
+                (throw (ex-info "Unverifiable credential"
+                                {:credential credential
+                                 :error :credential/unverifiable
+                                 :message e}))))
+      {:subject credential})))
```

Comment:
How does this work in practice? Are credentials automatically verified if there's no jws at the `["proof" "jws"]` path in the credential?

Reply:
That's right - if there's nothing to check, it's implicitly verified.

Reply:
Ok, so what's the plan for requiring verification (in non "open-api" ledgers)? How do we plan on rejecting queries to servers that require that the issuer has granted the user permission to query?

Reply:
That's not handled at this level - here we're only checking if a supplied json blob has a valid signature. If we get an issuer id then we can make policy decisions about what they can query/transact. I believe that's covered by issue #296

Reply:
Ok, I won't hold the pr up over this because I haven't seen the way the whole system works yet, but I would be way more comfortable if the `verify` function returned nil if there was nothing to verify.

Reply:
I'm fine with making it explicit, that's an easy change. :+1: